### ISSUE TITLE
0.15.22 regression: external requires are mapped to the main file

### ISSUE THREAD

**Issue by adamburgess** [systemjs/builder#648]
input file: app.js

```js
import React from 'react'
console.log(React);
```

**0.15.21** build to `cjs` with `react` as external: `jspm build app.js --format cjs --externals react`
```js
'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));
console.log(React);
//# sourceMappingURL=build.js.map
```
As you can see, it require's `react`, as expected.

**0.15.22** build to `cjs` with `react` as external: `jspm build app.js --format cjs --externals react`
```js
'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react/react.js'));
console.log(React);
//# sourceMappingURL=build.js.map
```
Here it erroneously require's `react/react.js`.

The bug is still there in 0.15.23.

**Comment by adamburgess** [comment]
so the fix in eabc810199b0a23fa284faae6f12a7b6e6083d99 wasn't perfect, it seems

**Comment by guybedford** [comment]
@adamburgess right, we should extend the function `getAlias` to do package detection and remove mains at https://github.com/systemjs/builder/blob/master/lib/utils.js#L77 before running the reverse map configuration.

The package lookup is provided at https://github.com/systemjs/builder/blob/master/lib/utils.js#L302 so this should be as simple as something like:

```javascript
var packageName = getPackage(loader.packages, loader.decanonicalize(canonicalName));
var packageMain = packageName && loader.packages[packageName].main;
if (packageMain && canonicalName.substr(canonicalName.length - packageMain.length - 1) == '/' + packageMain)
  canonicalName = canonicalName.substr(0, canonicalName.length - packageMain.length - 1);
```

**Comment by guybedford** [comment]
Added in https://github.com/systemjs/builder/commit/5e727251d63f0888a1062540102ef729b474b319.

**Comment by guybedford** [comment]
Released in 0.15.24.
